java/code/src/com/suse/oval/ovaldownloader/OVALDownloader.java

```diff
@@ -0,0 +1,178 @@
+ * maintainer to ensure the integrity and consistency of the consumed OVAL data.
+ * */
+public class OVALDownloader {
+    /**
+     * The path where downloaded OVAL files will be stored.
+     * TODO: Decide on a better location to cache OVAL files
+     * */
+    private static final String DOWNLOAD_PATH = "/var/log/rhn/ovals/";
+    /**
+     * A configuration object that corresponds to {@code oval.config.json}
+     * */
+    private final OVALConfig config;
+
+    /**
+     * Default constructor
+     *
+     * @param configIn A configuration object that corresponds to {@code oval.config.json}
+     * */
+    public OVALDownloader(OVALConfig configIn) {
+        this.config = configIn;
+    }
+
+    /**
+     * Download the OVAL data that correspond to the given {@code osFamily} and {@code osVersion} and cache
+     * them for future access.
+     *
+     * @param osFamily the family of the Operating system (OS) to download OVAL data for.
+     * @param osVersion the version of the Operating system (OS) to download OVAL data for.
+     * @return {@link OVALDownloadResult}
+     * */
+    public OVALDownloadResult download(OsFamily osFamily, String osVersion) throws IOException {
+        Optional<OVALSourceInfo> sourceInfoOpt = config.lookupSourceInfo(osFamily, osVersion);
+        if (sourceInfoOpt.isEmpty()) {
+            throw new IllegalArgumentException(
+                    String.format(
+                            "OVAL sources for '%s' '%s' is not configured. Please ensure OVAL is configured " +
+                                    "correctly oval.config.json",
+                            osFamily, osVersion));
+        }
+
+        return doDownload(sourceInfoOpt.get());
+    }
+
+    /**
+     * A helper method for download OVAL data.
+     * */
+    private OVALDownloadResult doDownload(OVALSourceInfo sourceInfo)
+            throws IOException {
+        OVALDownloadResult result = new OVALDownloadResult();
+
+        String vulnerabilityInfoSource = sourceInfo.getVulnerabilitiesInfoSource();
+        String patchInfoSource = sourceInfo.getPatchInfoSource();
+
+        if (StringUtils.isNotBlank(vulnerabilityInfoSource)) {
+            File vulnerabilityFile = downloadOVALFile(vulnerabilityInfoSource);
+            result.setVulnerabilityFile(vulnerabilityFile);
+        }
+
+        if (StringUtils.isNotBlank(patchInfoSource)) {
+            File patchFile = downloadOVALFile(patchInfoSource);
+            result.setPatchFile(patchFile);
+        }
+
+        return result;
+    }
+
+    private File downloadOVALFile(String vulnerabilityInfoSource) throws IOException {
+        URL vulnerabilityInfoURL = new URL(vulnerabilityInfoSource);
+        String vulnerabilityInfoOVALFilename = FilenameUtils.getName(vulnerabilityInfoURL.getPath());
+        File vulnerabilityFile =
+                new File(DOWNLOAD_PATH + vulnerabilityInfoOVALFilename);
```

Comment:
This will copy the file to a location under `/var/log/rhn/ovals/`. This is not a log file. If this is a permanent file we need to store we should define a better location. IF the file is temporary and will be deleted after processing, uyuni already have a cache/temporary location defined which can be used (I think is `/var/cache/rhn/`.

Reply:
A cache location would be ideal. However, I would leave it to the users to decide whether to delete the file after processing or not (through configuration). It can also be useful for debugging.

Reply:
Done. OVAL data is now cached under `/var/cache/rhn/ovals`.